components/kyma-environment-broker/common/hyperscaler/account_provider.go

```diff
@@ -41,13 +46,17 @@ func HyperscalerTypeForPlanID(planID string) (Type, error) {
 }
 
 func (p *accountProvider) GardenerCredentials(hyperscalerType Type, tenantName string) (Credentials, error) {
-
 	if p.gardenerPool == nil {
 		return Credentials{},
 			errors.New("failed to get Gardener Credentials. Gardener Account pool is not configured")
 	}
 
-	return p.gardenerPool.Credentials(hyperscalerType, tenantName)
+	secretBinding, err := p.gardenerPool.CredentialsSecretBinding(hyperscalerType, tenantName)
+	if err != nil {
+		return Credentials{}, err
```

Comment:
please always wrap an error:
```go
return ... errors.Wrap(err, "while ...")
```
not only here but everywhere where an error is returned

Reply:
Ayy ayy, capitan ⚓